### PR TITLE
Improve UI of MUXControlsTestApp

### PR DESCRIPTION
**Pull request by chingucoding**
## Description

This PR aims to improve the visuals and code of the MUXControlsTestApp
## Motivation and Context

Even though designed for tests, having devs use the MUXControlsTestApp should not be impossible or unpleasant due to unreadable text and other UI (and code) that is not optimal to use.
## How Has This Been Tested?

Started app. Tests should not be affected by this.
## Screenshots (if appropriate):

# Additional comments:
* NavigationView: "BlankPage" is unused. Should we remove it?
* ButtonInteractionPage and SliderInteractionPage are (at least for me) not visible in the MUXControlsTestApp. Are they expected to not be viewable?
* Pivot: The CommandBar on the test page does nothing, can it be removed?
* ScrollBar: The test page seems to have nothing rendered visible.
* ~ProgressBar: Since we now have the NumberBox, should we switch away from the "Set ... Value" buttons and instead use bindings?~ Interactiontests did not work, thus lets leave it the way it currently is
* Happy new year! 🎉

### CODE REVIEW

**Comment by mdtauk** [comment]
- Increase and Improve Spacing between control property controls;

- Increase Margins on pages and container regions;

- Proper separation between example controls and the property forms;

- Consistent layout for all test pages (perhaps include template XAML for developers and testers;

**Comment by chingucoding** [comment]
> * Increase and Improve Spacing between control property controls;
> * Increase Margins on pages and container regions;

This is something I already did on some pages,but not on all. If you find any pages that need improvement, feel free to tell me, and I will take a look 😊

> * Proper separation between example controls and the property forms;

I agree, however I don't know how you would do that without having pages contain weird layouts e.g. the NavigationView pages, where we use the NavigationView as root of the page. Do you have any ideas for this?

> * Consistent layout for all test pages (perhaps include template XAML for developers and testers;

Unfortunately, there are rarely any layouts/patterns that are used more than once or twice. This is due the fact that different controls have different ways to test them. So having a consistent layout among all pages might be quite difficult. But if you have an idea feel free to comment them :)

**Comment by mdtauk** [comment]
> > * Increase and Improve Spacing between control property controls;
> > * Increase Margins on pages and container regions;
> 
> This is something I already did on some pages,but not on all. If you find any pages that need improvement, feel free to tell me, and I will take a look 😊
> 
> > * Proper separation between example controls and the property forms;
> 
> I agree, however I don't know how you would do that without having pages contain weird layouts e.g. the NavigationView pages, where we use the NavigationView as root of the page. Do you have any ideas for this?
> 
> > * Consistent layout for all test pages (perhaps include template XAML for developers and testers;
> 
> Unfortunately, there are rarely any layouts/patterns that are used more than once or twice. This is due the fact that different controls have different ways to test them. So having a consistent layout among all pages might be quite difficult. But if you have an idea feel free to comment them :)


Its not about making each page the same layout, but the same separation between tests, properties, and the control example itself standing out from each other.

image

So it becomes obvious what the example is that you should be looking at.

What properties you can adjust.

What actions/tests you can perform

File: dev/Repeater/APITests/RepeaterTests.cs [hunk 1]
```diff
@@ -201,7 +201,7 @@ public void VerifyCurrentAnchor()
                 // Test is failing in chk configuration due to:
                 // Bug #1726 Test Failure: RepeaterTests.VerifyCurrentAnchor 
                 Log.Warning("Skipping test for Debug builds.");
-                return;
+                //return;
```
Comment (kaiguo) [inline dev/Repeater/APITests/RepeaterTests.cs]: Uncomment?

Comment (chingucoding) [reply dev/Repeater/APITests/RepeaterTests.cs]: Oh missed that completely, thanks for pointing that out!

**Comment by chingucoding** [comment]
> Its not about making each page the same layout, but the same separation between tests, properties, and the control example itself standing out from each other.

Thats a good point. I will see if what pages that need improvements in that area.

@kaiguo Thanks for starting azure!

**Comment by kaiguo** [comment]
@chingucoding, there's one more `AnimatedVisualPlayerTests.AccessibilityTest` failing, could you take a look 😊?

**Comment by chingucoding** [comment]
> @chingucoding, there's one more `AnimatedVisualPlayerTests.AccessibilityTest` failing, could you take a look 😊?

Sure :)

**Comment by chingucoding** [comment]
@kaiguo , I have reverted my changes for the AnimatedVIsualPlayer. Visual Studio did not want to run the test for the AnimatedVisualPlayer page, so I couldn't investigate the test failing.

**Comment by kaiguo** [comment]
> @kaiguo Kai Guo FTE , I have reverted my changes for the AnimatedVIsualPlayer. Visual Studio did not want to run the test for the AnimatedVisualPlayer page, so I couldn't investigate the test failing.

I looked at the log, seems like some PropertyChangedEventWaiter timed out, I'm not sure why adding a column would make it time out. Reverting sounds fine to me, I'll re-run the tests.

**Comment by chingucoding** [comment]
It seemed like that the test that was fixed with #1797 was failing, so I rebased this branch onto master. Hopefully that helps.

**Comment by StephenLPeters** [comment]
> > > * Increase and Improve Spacing between control property controls;
> > > * Increase Margins on pages and container regions;
> > 
> > 
> > This is something I already did on some pages,but not on all. If you find any pages that need improvement, feel free to tell me, and I will take a look 😊
> > > * Proper separation between example controls and the property forms;
> > 
> > 
> > I agree, however I don't know how you would do that without having pages contain weird layouts e.g. the NavigationView pages, where we use the NavigationView as root of the page. Do you have any ideas for this?
> > > * Consistent layout for all test pages (perhaps include template XAML for developers and testers;
> > 
> > 
> > Unfortunately, there are rarely any layouts/patterns that are used more than once or twice. This is due the fact that different controls have different ways to test them. So having a consistent layout among all pages might be quite difficult. But if you have an idea feel free to comment them :)
> 
> Its not about making each page the same layout, but the same separation between tests, properties, and the control example itself standing out from each other.
> 
> image
> 
> So it becomes obvious what the example is that you should be looking at.
> 
> What properties you can adjust.
> 
> What actions/tests you can perform

I've actually been wishing/hoping for this for a long time.  I think the best example of a good test page whose layout could be the base line is the Scroller's Exercise Scroller API page which @RBrid wrote.  The teaching tip, exercise swipe control api's, and RadioButtons test pages were built with this in mind. Personally I would love if all the pages got updated to look like this.

**Comment by chingucoding** [comment]
> I've actually been wishing/hoping for this for a long time. I think the best example of a good test page whose layout could be the base line is the Scroller's Exercise Scroller API page which @RBrid wrote. The teaching tip, exercise swipe control api's, and RadioButtons test pages were built with this in mind. Personally I would love if all the pages got updated to look like this.

Agreed, those pages are better to use. I will try to get all pages into this pattern/layout. 

However, on some pages I am not sure what to do since they have a lot of controls e.g. the ItemsRepeater overview page. Should I just propose changes there, and you just look if it makes sense? Or would it be better to update the questions at the top of this thread?

**Comment by ranjeshj** [comment]
ItemsRepeater pages are purely for trying things out. They are not used by interaction tests so you can freely update them. Much of the other pages have interaction tests driving their pages, so you want to make sure atleast the UIA tree does not get changed much or be willing to update the tests as well.

**Comment by chingucoding** [comment]
> ItemsRepeater pages are purely for trying things out. They are not used by interaction tests so you can freely update them. Much of the other pages have interaction tests driving their pages, so you want to make sure atleast the UIA tree does not get changed much or be willing to update the tests as well.

I see, thank you for the clarification!

**Comment by karenbtlai** [comment]
> > > * Increase and Improve Spacing between control property controls;
> > > * Increase Margins on pages and container regions;
> > 
> > 
> > This is something I already did on some pages,but not on all. If you find any pages that need improvement, feel free to tell me, and I will take a look 😊
> > > * Proper separation between example controls and the property forms;
> > 
> > 
> > I agree, however I don't know how you would do that without having pages contain weird layouts e.g. the NavigationView pages, where we use the NavigationView as root of the page. Do you have any ideas for this?
> > > * Consistent layout for all test pages (perhaps include template XAML for developers and testers;
> > 
> > 
> > Unfortunately, there are rarely any layouts/patterns that are used more than once or twice. This is due the fact that different controls have different ways to test them. So having a consistent layout among all pages might be quite difficult. But if you have an idea feel free to comment them :)
> 
> Its not about making each page the same layout, but the same separation between tests, properties, and the control example itself standing out from each other.
> 
> image
> 
> So it becomes obvious what the example is that you should be looking at.
> 
> What properties you can adjust.
> 
> What actions/tests you can perform

Agreed, this is definitely a huge improvement. Thanks for doing this :)

**Comment by chingucoding** [comment]
Looks like ProgressBar and TextCommandBarFlyoutTests are/were being reported as failing. Was this related to the AutoSuggestBoxTest? 

What is the "Work Item" indicated in the log?

Link to Pipeline build:
https://dev.azure.com/ms/microsoft-ui-xaml/_build/results?buildId=56834&view=results

**Comment by StephenLPeters** [comment]
> Looks like ProgressBar and TextCommandBarFlyoutTests are/were being reported as failing. Was this related to the AutoSuggestBoxTest?
> 
> What is the "Work Item" indicated in the log?
> 
> Link to Pipeline build:
> https://dev.azure.com/ms/microsoft-ui-xaml/_build/results?buildId=56834&view=results

I think the work item here refers to the helix batch of work that the failing test was a part of. @kmahone is that right?

**Comment by StephenLPeters** [comment]
@chingucoding These test failures look like real issue to me, will you be able to take a look?

**Comment by chingucoding** [comment]
> @chingucoding These test failures look like real issue to me, will you be able to take a look?

Since the tests are not failing on my machine for some reason, I mostly reverted the pages in question as they are usable and having the tests be reliable is most important.

**Comment by chingucoding** [comment]
Since the last CI run was successful, I think it is the best to merge this PR now, as a lot of pages have been improved. I have merged the latest commits from master in this branch to resolve potential merge problems.

If the CI build started by @ranjeshj finishes and is successful, I think it is best to merge this PR and make a separate PR for additional improvements.

Regarding latter, should I create an issue for this to collect items not covered by this PR?

**Comment by ranjeshj** [comment]
> Since the last CI run was successful, I think it is the best to merge this PR now, as a lot of pages have been improved. I have merged the latest commits from master in this branch to resolve potential merge problems.
> 
> If the CI build started by @ranjeshj finishes and is successful, I think it is best to merge this PR and make a separate PR for additional improvements.
> 
> Regarding latter, should I create an issue for this to collect items not covered by this PR?

Sure. We please do a follow up PR if it is getting too painful to merge.

**Comment by chingucoding** [comment]
> Sure. Please do a follow up PR if it is getting too painful to merge.

If this is something, the WinUI team think is worth investing more time, sure. 

Currently I do not have too much time to work further on this, so instead of having to continuously merge master into this PR, it is easier to merge this.

**Comment by StephenLPeters** [comment]
I would say merge it in, even if you don't plan on doing other test page updates these are still valuable.  We will probably not get to this in the foreseeable future. I'll let @ranjeshj make the call though.

**Comment by ranjeshj** [comment]
> I would say merge it in, even if you don't plan on doing other test page updates these are still valuable. We will probably not get to this in the foreseeable future. I'll let @ranjeshj make the call though.

I agree. This already cleans up a bunch and we can incrementally improve this. Thanks @chingucoding for making these updates.

**Pull request merged**